### PR TITLE
Feature/APP-1431: IAS integration

### PR DESCRIPTION
**Pull request by Iliaspit**

### CODE REVIEW

File: src/Bling.js [hunk 1]
```diff
@@ -125,6 +125,8 @@ class Bling extends Component {
             PropTypes.bool,
             PropTypes.array
         ]),
+        withIAS: PropTypes.func,
+        iaspPubId: PropTypes.string,
```
Comment (willhowlett) [inline src/Bling.js]: `iaspPubId`shouldnt be needed here right? It can just be set in the callback function

File: src/Bling.js [hunk 2]
```diff
@@ -608,8 +610,25 @@ class Bling extends Component {
...
 
     renderAd() {
+        const { withIAS, iaspPubId, adUnitPath } = this.props;
+        const divId = this._divId;
+        const slotSize = this.getSlotSize();
+
         this.defineSlot();
-        this.display();
+
+        if (withIAS) {
```
Comment (willhowlett) [inline src/Bling.js]: can we rename `withIas` to something like `displayCallback`? Then replace this with 

```
if (displayCallback) {
  displayCallback({
    display: this.display,
    adUnitPath,
    slotSize,
    divId,
  })
}
```
(we could add any other vars that might be needed by other callbacks as well, I'm just putting what is used here)

This should allow us to use this.display as the callback directly within the ad component, e.g. something like:
```
      <GPT
        displayCallback={({
          display,
          adUnitPath,
          slotSize,
          divId,
        }) => {
          withIAS({
            diplayAdsCb: display,
            pubId: iaspPubId,
            adSlotData: {
              adUnitPath,
              sizes: slotSize,
              adSlotId: divId,
            }
          })
        }}
      />
```

Does that make sense? Basically just don't have reference to withIas in react-gpt

File: src/Bling.js (hunk 1 above)
Comment (Iliaspit) [reply src/Bling.js]: umm i think this is passed from the ap asp a prop right? don't we need to pass it as a prop to the callback?

Comment (willhowlett) [reply src/Bling.js]: sorry ignore that comment - the comment below addresses it

File: src/Bling.js (hunk 2 above)
Comment (Iliaspit) [reply src/Bling.js]: yeah I could do that!

Comment (willhowlett) [reply src/Bling.js]: woo (can you document it in https://github.com/dennisinteractive/react-gpt/blob/master/docs/api/ReactGPT.md#Bling too - I figure if we're going to be using this forked version for the time being we should keep it clean and usable)

**Review by willhowlett — COMMENTED** [review]
Awesome. Please can you update the readme to document the new prop functionality

File: src/Bling.js [hunk 3]
```diff
@@ -608,8 +609,24 @@ class Bling extends Component {
...
+
         this.defineSlot();
-        this.display();
+
+        if (displayCallback) {
+          displayCallback({
+            display: this.display,
+            adSlotData: {
+              adUnitPath,
+              sizes: slotSize,
```
Comment (willhowlett) [inline src/Bling.js]: this should probably just be called `size` right? (it's just one slot size)

File: lib/Bling.js [hunk 4]
```diff
@@ -431,18 +431,39 @@ var Bling = (_temp2 = _class = function (_Component) {
...
 
             if (content) {
                 Bling._adManager.googletag.content().setContent(adSlot, content);
             } else {
                 if (!Bling._adManager._disableInitialLoad && !Bling._adManager._syncCorrelator) {
                     Bling._adManager.updateCorrelator();
                 }
-                Bling._adManager.googletag.display(divId);
+
+                if (displayCallback) {
```
Comment (willhowlett) [inline lib/Bling.js]:
```suggestion
                if (typeof displayCallback === "function") {
```

Comment (Iliaspit) [reply lib/Bling.js]: ah thanks was going to make that change

File: src/Bling.js (hunk 2 above)
Comment (willhowlett) [reply src/Bling.js]: this callback appears to be working nicely from my local usage. Please can it's usage get added to the readme as per above comment?

File: README.md [hunk 5]
```diff
@@ -111,6 +111,13 @@ class Application extends React.Component {
 }
 ```
 
+
```
Comment (willhowlett) [inline README.md]: This is the wrong place for this info. We don't need to specify how we've used the callback in our sites, we just need to document the usage of the displayCallback prop here: https://github.com/dennisinteractive/react-gpt/blob/master/docs/api/ReactGPT.md#Bling

Comment (Iliaspit) [reply README.md]: of course - how silly, they do proper docs. Too long in Dennis

Comment (willhowlett) [reply README.md]: DELETE THIS BIT THEN ILIAS OMG

Comment (willhowlett) [reply README.md]: WE ONLY NEED THE DOCS IN ReactGPT.md

Comment (willhowlett) [reply README.md]: OMG

Comment (willhowlett) [reply README.md]: OMG ILIAS

Comment (willhowlett) [reply README.md]: OMFG

Comment (Iliaspit) [reply README.md]: haha okay done

Comment (willhowlett) [reply README.md]: FOMFGFFFF

**Review by willhowlett — APPROVED** [review]

**Pull request merged**